File: pronacej-backend-seguridad/src/main/java/net/latinus/sistema/integral/gestion/seguridad/service/IA/SeguimientoService.java
package net.latinus.sistema.integral.gestion.seguridad.service.IA;

import jakarta.servlet.http.HttpServletRequest;
import net.latinus.sistema.integral.gestion.seguridad.model.both.BodyEncriptado;
import net.latinus.sistema.integral.gestion.seguridad.model.both.ia.SeguimientoConductualDTO;
import net.latinus.sistema.integral.gestion.seguridad.model.both.ia.SeguimientoPsicologicoDTO;
import net.latinus.sistema.integral.gestion.seguridad.model.response.PaginacionResponse;
import net.latinus.sistema.integral.gestion.seguridad.model.response.RespuestaPorDefectoAuditoria;

public interface SeguimientoService {

    // region Seguimiento Psicologico

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data EncuestaDTO
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado request peticion.
     * @return RespuestaPorDefectoAuditoria<SeguimientoPsicologicoDTO>
     */
    RespuestaPorDefectoAuditoria<PaginacionResponse<SeguimientoPsicologicoDTO>> obtenerSeguimientosPsicologicos(HttpServletRequest httpServletRequest,
                                                                                                                BodyEncriptado bodyEncriptado);

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data EncuestaDTO si la
     * encuesta se creo con exito
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado        objeto SeguimientoPsicologicoDTO.
     * @return RespuestaPorDefectoAuditoria<Boolean>
     */
    RespuestaPorDefectoAuditoria<Boolean> crearSeguimientoPsicologico(HttpServletRequest httpServletRequest,
                                                                           BodyEncriptado bodyEncriptado);


    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data EncuestaDTO si la
     * empresa se creo con exito
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado        objeto SeguimientoPsicologicoDTO.
     * @return RespuestaPorDefectoAuditoria<Boolean>
     */
    RespuestaPorDefectoAuditoria<Boolean> actualizarSeguimientoPsicologico(HttpServletRequest httpServletRequest,
                                                                           BodyEncriptado bodyEncriptado);

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data EncuestaDTO si la
     * empresa se creo con exito
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado        objeto SeguimientoPsicologicoDTO.
     * @return RespuestaPorDefectoAuditoria<Boolean>
     */
    RespuestaPorDefectoAuditoria<Boolean> eliminarSeguimientoPsicologico(HttpServletRequest httpServletRequest,
                                                                         BodyEncriptado bodyEncriptado);

    // endregion

    // region Seguimiento Conductual

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data EncuestaDTO
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado request peticion.
     * @return RespuestaPorDefectoAuditoria<SeguimientoConductualDTO>
     */
    RespuestaPorDefectoAuditoria<PaginacionResponse<SeguimientoConductualDTO>> obtenerSeguimientosConductuales(HttpServletRequest httpServletRequest,
                                                                                                               BodyEncriptado bodyEncriptado);

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data EncuestaDTO si la
     * encuesta se creo con exito
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado        objeto SeguimientoConductualDTO.
     * @return RespuestaPorDefectoAuditoria<Boolean>
     */
    RespuestaPorDefectoAuditoria<Boolean> crearSeguimientoConductual(HttpServletRequest httpServletRequest,
                                                                      BodyEncriptado bodyEncriptado);


    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data EncuestaDTO si la
     * empresa se creo con exito
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado        objeto SeguimientoConductualDTO.
     * @return RespuestaPorDefectoAuditoria<Boolean>
     */
    RespuestaPorDefectoAuditoria<Boolean> actualizarSeguimientoConductual(HttpServletRequest httpServletRequest,
                                                                           BodyEncriptado bodyEncriptado);

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data EncuestaDTO si la
     * empresa se creo con exito
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado        objeto SeguimientoConductualDTO.
     * @return RespuestaPorDefectoAuditoria<Boolean>
     */
    RespuestaPorDefectoAuditoria<Boolean> eliminarSeguimientoConductual(HttpServletRequest httpServletRequest,
                                                                         BodyEncriptado bodyEncriptado);
    // endregion
}

File: pronacej-front/src/app/modules/administracion/services/seguimiento.service.ts
import { Injectable } from '@angular/core';
import { SeguimientoConductualDTO } from 'app/core/model/both/ia/seguimientoConductualDTO.model';
import { SeguimientoPsicologicoDTO } from 'app/core/model/both/ia/seguimientoPsicologicoDTO.model';
import { PaginacionRequest } from 'app/core/model/request/PaginacionRequest.model';
import { PaginacionResponse } from 'app/core/model/response/PaginacionResponse.model';
import { RespuestaPorDefecto } from 'app/core/model/response/RespuestaPorDefecto.model';
import { BackendService } from 'app/core/services/backend.service';
import { Observable } from 'rxjs';

@Injectable({
    providedIn: 'root',
})
export class SeguimientoService {
    private path = '/seguimiento';

    constructor(private backendService: BackendService) {}

    obtenerSeguimientosPsicologicos(
        paginacionRequest: PaginacionRequest,
        nemonicoMenu: string
    ): Observable<
        RespuestaPorDefecto<PaginacionResponse<SeguimientoPsicologicoDTO>>
    > {
        let endPoint = this.path + '/obtenerSeguimientosPsicologicos';
        return this.backendService.postFinal(
            endPoint,
            paginacionRequest,
            nemonicoMenu
        );
    }

    crearSeguimientoPsicologico(
        psicologicoDTO: SeguimientoPsicologicoDTO,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<Boolean>> {
        let endPoint = this.path + '/crearSeguimientoPsicologico';
        return this.backendService.postFinal(
            endPoint,
            psicologicoDTO,
            nemonicoMenu
        );
    }

    actualizarSeguimientoPsicologico(
        psicologicoDTO: SeguimientoPsicologicoDTO,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<Boolean>> {
        let endPoint = this.path + '/actualizarSeguimientoPsicologico';
        return this.backendService.postFinal(
            endPoint,
            psicologicoDTO,
            nemonicoMenu
        );
    }

    eliminarSeguimientoPsicologico(
        psicologicoDTO: SeguimientoPsicologicoDTO,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<Boolean>> {
        let endPoint = this.path + '/eliminarSeguimientoPsicologico';
        return this.backendService.postFinal(
            endPoint,
            psicologicoDTO,
            nemonicoMenu
        );
    }

    obtenerSeguimientosConductuales(
        paginacionRequest: PaginacionRequest,
        nemonicoMenu: string
    ): Observable<
        RespuestaPorDefecto<PaginacionResponse<SeguimientoConductualDTO>>
    > {
        let endPoint = this.path + '/obtenerSeguimientosConductuales';
        return this.backendService.postFinal(
            endPoint,
            paginacionRequest,
            nemonicoMenu
        );
    }

    crearSeguimientoConductual(
        conductualDTO: SeguimientoConductualDTO,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<Boolean>> {
        let endPoint = this.path + '/crearSeguimientoConductual';

        return this.backendService.postFinal(
            endPoint,
            conductualDTO,
            nemonicoMenu
        );
    }

    actualizarSeguimientoConductual(
        conductualDTO: SeguimientoConductualDTO,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<Boolean>> {
        let endPoint = this.path + '/actualizarSeguimientoConductual';
        return this.backendService.postFinal(
            endPoint,
            conductualDTO,
            nemonicoMenu
        );
    }

    eliminarSeguimientoConductual(
        conductualDTO: SeguimientoConductualDTO,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<Boolean>> {
        let endPoint = this.path + '/eliminarSeguimientoConductual';
        return this.backendService.postFinal(
            endPoint,
            conductualDTO,
            nemonicoMenu
        );
    }
}
